src/components/explain-states/utils/to-strict-ejson/index.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * This module was contributed by Manuel Fontan (manuel.fontan@mongodb.com)
```

Comment:
Does [`mongodb-query-parser`](https://github.com/mongodb-js/query-parser) not handle these cases? If it does not, let's create an issue there so they will be fixed across all modules. @rueckstiess perhaps you've come across these in `ejson-shell-parser`?